### DISCUSSION TITLE
Pymetis for weighted graph

### DISCUSSION THREAD

**Discussion by abhisheksharma1395** [inducer/pymetis#39] [Q&A]
Hi, 
I am a university student. I am doing some research project which requires me to use pyMetis.
I am having a hard time using pymetis, and I wasn't able to find any useful resources or examples on the web.
I wanted to partition a  weighted graph as a part of my requirement, which I built via networkx.
I have visited https://documen.tician.de/pymetis/_modules/pymetis.html#part_graph but i find it hard to understand without example. 
Can you help me with this or redirect me to any useful resource.

Regards

**Comment by inducer** [comment]
http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/manual.pdf

**Reply by abhisheksharma1395** [reply]
Thank You

**Reply by liubo-cs** [reply]
Hi，
Greetings! Can you provide a concrete example of the eweights? For example, I am confused about
(1) what is the format of the eweights? a numpy array, or a python list?
(2) how to order the edge list? I am completely confused here.
Take the example of https://github.com/inducer/pymetis, where I tried all the following of eweights (in total there are 11 edges, so m=11):

# in total 11 edges
# First attempt:
eweights_list = np.ones(22)
# Second attempt:
eweights_list = list(np.ones(22))
# Third attempt:
eweights_list = [np.ones(3),
                  np.ones(3),
                  np.ones(4),
                  np.ones(4),
                  np.ones(3),
                  np.ones(3),
                  np.ones(2)]

And then call
n_cuts, membership = pymetis.part_graph(2, adjacency=adjacency_list, eweights = eweights_list)

But none of them work. Appreciate your clarification! An example would greatly help. Thanks!

**Reply by jlarsen-usgs** [reply]
@inducer 
I found the single example problem, as well as the documentation for pymetis, inadequate as well. Fortunately, I figured out how to do a vertex weighted part graph. Please consider updating your docstings/docs, and examples to give a clear explanation on how to use the additional keyword arguments such as `vweights` and `eweights`. 

In my personal opinion, this should be an issue, because there is insufficient documentation for all of the options in this python wrapper around METIS. Also, the link to the METIS manual that you've shared doesn't work. For folks that are interested, you can find it here: [metis manual](https://github.com/inducer/pymetis/blob/main/src/metis/manual/manual.pdf)

**Reply by inducer** [reply]
Your contribution to make improvements along these lines would be welcome!
